qualtran/qref_interop/_bloq_to_qref.py

```diff
@@ -0,0 +1,311 @@
+            # - second one expands composite register (which have no counterpart in Bartiq) into
+            #   required number of single ports.
+            PortV1(
+                name=expanded_name,
+                direction=direction,
+                size=reg.bitsize if isinstance(reg.bitsize, int) else str(reg.bitsize),
+            )
+            for flat_name, direction in _names_and_dir_from_register(reg)
+            for expanded_name in _expand_name_if_needed(flat_name, reg.shape)
+        ],
+        key=lambda p: p.name,
+    )
+
+
+def _opposite(direction: str) -> str:
+    return "out" if direction == "in" else "in"
+
+
+def _relative_port_name(soquet: Soquet, direction) -> str:
+    """Given a Soquet and direction, determine the relative name of corresponding Bartiq Port.
+
+    The relative name is always computed wrt. the parent RoutineV1.
+    This function correctly recognizes the fact, that in any connection, the input parent
+    ports serve as outputs for the connection (and vice versa).
+    """
+    if len(soquet.idx) > 1:
+        raise NotImplementedError(
+            "Soquets referencing more than one index in composite register are not yet supported. "
+            f"The error was caused by the following soquet: {soquet}."
+        )
+    # We add another suffix iff soquet references idx in composite register
+    suffix = f"_{soquet.idx[0]}" if soquet.idx else ""
+    return (
+        # If soquet references BlogqInstance, the corresponding object in Bartiq
+        # references child - construct dotted relative name.
+        # Otherwise, soquet references the parent port, and so for the output direction,
+        # the port is in_parent_port, which is why we include _opposte here.
+        f"{_bloq_instance_name(soquet.binst)}.{direction}_{soquet.reg.name}{suffix}"
+        if isinstance(soquet.binst, BloqInstance)
+        else f"{_opposite(direction)}_{soquet.reg.name}{suffix}"
+    )
+
+
+def _import_connection(connection: QualtranConnection) -> dict[str, Any]:
+    """Import connection from Qualtran."""
+    return {
+        "source": _relative_port_name(connection.left, "out"),
+        "target": _relative_port_name(connection.right, "in"),
+    }
+
+
+def _import_resources(bloq: Bloq) -> list[dict[str, Any]]:
+    """Import resources from Bloq's t_complexity method."""
+    t_complexity = bloq.t_complexity()
+    resource_names = ["t", "clifford", "rotations"]
+    resources = []
+    for name in resource_names:
+        resources.append(
+            {
+                "name": name,
+                "value": _ensure_primitive_type(getattr(t_complexity, name)),
+                "type": "additive",
+            }
+        )
+    return resources
+
+
+def _ensure_primitive_type(value: Any) -> Union[int, float, str, None]:
+    """Ensure given value is of primitive type (e.g. is not a sympy expression)."""
+    return value if value is None or isinstance(value, (int, float, str)) else str(value)
```

Comment:
If the goal is to check whether the value is a symbolic or not; you can consider using the `is_symbolic` method in `qualtran.symbolics` -- it may be more consistent since `isinstance` check would fail for types like `numpy.int64`.